packages/web3torrent/src/pages/file/File.tsx

```diff
@@ -31,21 +31,27 @@ interface Props {
 
 const File: React.FC<Props> = props => {
   const web3Torrent = useContext(Web3TorrentContext);
-  const [torrent, setTorrent] = useState(parseMagnetURL(useLocation().hash));
+  const {infoHash} = useParams();
+  const queryParams = useQuery();
+  const [torrent, setTorrent] = useState<Torrent>(EmptyTorrent);
   const [, setPeers] = useState({});
   const [loading, setLoading] = useState(false);
   const [buttonLabel, setButtonLabel] = useState('Start Download');
   const [errorLabel, setErrorLabel] = useState('');
 
   const getLiveData = getTorrentAndPeersData(web3Torrent, setTorrent, setPeers);
 
+  useEffect(() => {
```

Comment:
Do we need to `useEffect` here? As written, this will run only on first render and will initialize `torrent` to `parseURL(infoHash, queryParams)`.

Reply:
It was the result of me discovering [this](https://statechannels.slack.com/archives/CMPHW40R0/p1586807045141000).
Said that, I will change that with
```const [torrent, setTorrent] = useState<Torrent>(() => parseURL(infoHash, queryParams));```

Which avoids the re-running of the function on each render, and it's cleaner IMO.

Reply:
is `parseURL` an expensive call? Also, not sure the above will work. It will set the `torrent` variable to a parameterless function (and not invoke the function). Unless my understanding of `useState` is flawed...

Reply:
[Might have answered my own question](https://reactjs.org/docs/hooks-reference.html#functional-updates)

Reply:
It works fine on my PC, and the tests pass. And according to the [god-awful docs](https://en.reactjs.org/docs/hooks-reference.html#lazy-initial-state) (that I don't know why they put the `useState` documentation in 2 different places, I didn't like this entire quirk of the useState API), it _should_ work, I think.